frontend/src/contexts/HostContext/HostReducer.js

```diff
@@ -163,21 +161,29 @@ function removeSubmittedCards(state) {
 function disconnectPlayer(state, { playerId }) {
   const removingCzar = state.players[playerId]?.isCzar;
 
+  // add conditional to check if the player is disconnecting due to duplicate name
+  if (!state.playerIDs.includes(playerId)) {
+    return state;
+  }
+
   const newState = {
     ...state,
     players: {
       ...state.players,
       [playerId]: {
         ...state.players[playerId],
-        isConnected: false,
-        isPlaying: false,
+        status: 'disconnected',
       },
     },
   };
 
   if (
     removingCzar &&
-    state.playerIDs.filter((id) => state.players[id].isConnected).length > 1 &&
+    state.playerIDs.filter(
+      (id) =>
+        state.players[id].status === 'playing' ||
+        state.players[id].status === 'staging',
```

Comment:
We could use a `.includes` here to clean this up:

`['playing', 'staging'].includes(state.players[id].status`